### PR TITLE
Propose build-std as a 2025H1 goal

### PR DESCRIPTION
**Pull request by davidtwco**
I've discussed this goal with the t-compiler/libs leads and attended the most recent t-cargo meeting to discuss this with t-cargo.

This goal shouldn't be merged straight away, so that the relevant stakeholders and teams have a chance to leave some feedback, but it can be merged before the goal opening deadline has lapsed.

cc @rust-lang/cargo (relevant team)
cc @wesleywiser @m-ou-se @Amanieu (stakeholders)

### CODE REVIEW

File: src/2025h1/build-std.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+*Arm's Rust team is @davidtwco, @adamgemmell, @jacobbramley, @JamieCunliffe and @Jamesbarford. This
+goal will be primarily worked on by @adamgemmell, but @davidtwco can always be contacted for
+updates.*
+
+## Summary
+
+Write an RFC for a minimum viable product (MVP) of build-std which has the potential to be
+stabilised once implemented (as opposed to the currently implemented MVP which is only suitable for
+experimentation and testing), and then implement it.
```
Comment (nikomatsakis) [inline src/2025h1/build-std.md:17]: Is the goal to *stabilize* in 6 months or just to write an RFC? The text seems to imply the former, but the "ownership and team asks" requests stabilization support from Cargo team.

Comment (davidtwco) [reply src/2025h1/build-std.md:17]: It's very unlikely that we'd stabilise this in six months, I expect we'd probably have the RFC completed in the first six months and perhaps have started on implementation. In the text you've commented on, I'm just trying to say that the MVP we produce should be one that *could* be stabilised, as opposed to the existing MVP of build-std, for example. I can cut some of the rows from the table at the bottom for the parts that we're unlikely to get to in this goal cycle.

Comment (dev-ardi) [reply src/2025h1/build-std.md:17]: What are the plans for goals that might take 2/3 cycles but not long enough to take many years?

File: src/2025h1/build-std.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+| Subgoal                                        | Owner(s) or team(s)  | Notes |
+| ---------------------------------------------- | -------------------- | ----- |
+| Discussion and moral support                   | ![Team][] [cargo]    |       |
+| Stabilise MVP of build-std                     | @adamgemmell         |       |
+| ↳ Author RFC                                   | @adamgemmell         |       |
+| ↳ Implementation                               | @adamgemmell         |       |
+| ↳ Standard reviews                             | ![Team][] [cargo]    |       |
+| ↳ RFC decision                                 | ![Team][] [cargo]    |       |
+| ↳ Author stabilization report                  | @adamgemmell         |       |
+| ↳ Stabilization decision                       | ![Team][] [cargo]    |       |
```
Comment (nikomatsakis) [inline src/2025h1/build-std.md:127]:
```suggestion
| Subgoal                                        | Owner(s) or team(s)  | Notes |
| ---------------------------------------------- | -------------------- | ----- |
| Discussion and moral support                   | ![Team][] [cargo]    |       |
| Stabilise MVP of build-std                     | @adamgemmell         |       |
| ↳ Author RFC                                   | @adamgemmell         |       |
| ↳ Implementation                               | @adamgemmell         |       |
| ↳ Standard reviews                             | ![Team][] [cargo]    |       |
```

I'm weakening these goals based on what you wrote above. My assumption is that your plan is to have the RFC *opened* (not decided upon) and implementation to have begun. Therefore you are not expecting the cargo team to reach an RFC decision or a stabilization decision.

File: src/2025h1/build-std.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  cases that build-std is intended to solve.
+
+  It is hoped that this RFC should demonstrate a thorough understanding of the design space
+  of build-std and give the responsible upstream teams confidence in our ownership of this
+  feature, and enabling those teams to make a fully informed decision on any proposals made.
+
+- Next, after and conditional on acceptance of this RFC, we will proceed with its
+  implementation.
+
+The fifty-thousand foot view of the work involved in this feature is:
```
Comment (nikomatsakis) [inline src/2025h1/build-std.md:81]: I'd probably move this to the "shiny future" section. I wonder if we should rename that to "the larger plan"

File: src/2025h1/build-std.md (hunk 1 above)
Comment (nikomatsakis) [reply src/2025h1/build-std.md:17]: Yes, that's fine, I am referring specifically to the "team asks" which included items like "stabilization decision". I don't think you are asking for that in this 6 month span.

**Review by nikomatsakis — CHANGES_REQUESTED** [review]

File: src/2025h1/build-std.md (hunk 2 above)
Comment (davidtwco) [reply src/2025h1/build-std.md]: Fixed

File: src/2025h1/build-std.md (hunk 3 above)
Comment (davidtwco) [reply src/2025h1/build-std.md]: Moved part of what followed from the fifty thousand foot view into the _Motivation_ and some into the _The status quo_.

**Review by nikomatsakis — APPROVED** [review]

**Pull request merged**